src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -198,11 +196,9 @@ public function set($id, $service, $scope = self::SCOPE_CONTAINER)
         $id = strtolower($id);
 
         if ('service_container' === $id) {
-            // BC: 'service_container' is no longer a self-reference but always
-            // $this, so ignore this call.
-            // @todo Throw InvalidArgumentException in next major release.
-            return;
+            throw new InvalidArgumentException('You cannot set service "service_container".');
```

Comment:
Hm, do we actually need this kind of special service at all or could we get rid of it?